CHANGELOG.md

```diff
@@ -4,8 +4,9 @@ dev
  - Fix bug when attempting to parse function with a magic scope parameter type hint (eg `function (self $param) { ... }`.
  - Implement `Analysis\TolerantExpressionAnalyser` which will convert analysis exceptions into the *mixed* type.
  - Add `Providers\DSL\Compilation\Parameters\ParameterCollection::contains` to check whether the collection contains a parameter.
- - Add `Providers\DSL\Compilation\Parameters\ParameterCollection::remove` to remove a previously added parameter.\
+ - Add `Providers\DSL\Compilation\Parameters\ParameterCollection::remove` to remove a previously added parameter.
  - Introduce `Analysis\INativeType::TYPE_NUMERIC` acting as a union type for `Analysis\INativeType::TYPE_INT` and `Analysis\INativeType::TYPE_DOUBLE`  
+ - Change identity hash to hash instances of `DateTimeInterface` by class and timestamp hence treating it as a value type throughout the PINQ API
```

Comment:
@TimeToogo the problem is that you are taking an opinionated approach on something that has different opinions all across the interwebz.

We decided to go with a by-ref check to avoid the problem overall until we have a way to solve it.

DBAL types are a semi-working solution (though static and opinionated).

In general, Value Objects would be the way to go, with a way to serialize/unserialize for in-memory and in-sql comparison.